returnn/torch/data/pipeline.py

```diff
@@ -383,6 +403,21 @@ def __iter__(self):
     def __getitem__(self, index):
         raise Exception(f"{self.__class__.__name__}.__getitem__ is not supported")
 
+    def set_seed(self, seed: int) -> BucketOrderingIterDataPipe:
+        """
+        Sets the seed for the next invocation of ``__iter__``, for compatibility with
+        ``torch.utils.data.graph_settings.apply_random_seed``.
+        """
+        self._seed = seed % (2**32)  # seed must be within [0, 2**32) for seeding RandomState
+        return self
+
+    def reset(self):
+        """resets the internal state of the data pipe"""
+        if self._seed is None:
+            self._seed = int(torch.empty((), dtype=torch.int32).random_().item())
```

Comment:
I wonder, `torch.utils.data.datapipes.iter.Shuffler` uses `int(torch.empty((), dtype=torch.int64).random_().item())`, i.e. 64 bit. Why do you restrict this to 32 bit now? I'm not sure it really matters much for us, but if there is no good reason for this, I would keep it as 64 bit.

Reply:
Numpy’s RandomState can only be seeded with numbers up to 2^32.

Reply:
Ok, it's fine because we need it between 0 and 2**32 - 1.